contrib_bots/lib/hangman.py

```diff
@@ -0,0 +1,167 @@
+                    end = "\n\n:crying_cat_face: Oh no, you lost! The " \
+                        "correct word was " + str(bot_self.currentword) + "." \
+                          " To play again, type `@hangman new`."
+                    bot_self.ended = True
+            stages = ["```", "```\n=============",
+                      ("```\n           |\n           |\n           |" +
+                       "\n           |\n           |\n============="),
+                      ("```\n   +-------+\n           |\n           |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n           |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n     " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n   | " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /| " +
+                       "      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n           |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n  /        |\n           |\n============="),
+                      ("```\n   +-------+\n   |       |\n   0       |\n  /|" +
+                       "\      |\n  / \      |\n           |\n=============")]
+            outcur = []
+            wrongguess = []
+            for i in range(0, len(bot_self.currentword)):
+                if bot_self.currentword[i] in bot_self.guessed:
+                    outcur.append(str(bot_self.currentword[i]))
+                else:
+                    outcur.append("_")
+            if "_" not in outcur:
+                end = "\n\n:party_popper: Congratulations! You won! " \
+                    "To play again, type @hangman new. :party_popper:"
+                bot_self.ended = True
+            for i in range(0, len(bot_self.guessed)):
+                if bot_self.guessed[i] not in bot_self.currentword:
+                    wrongguess.append(str(bot_self.guessed[i]))
+            bot_self.output = (stages[bot_self.stage] + "\n```\n\n```" +
+                               "  ".join(outcur) + "```\n\n```Wrong guesses:" +
+                               " " + ", ".join(wrongguess) + "```" + end)
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content'].lower()
+        original_content_split = original_content.split(" ")
+        if original_content_split[1] == "new":
+            HangmanHandler.get_new_word(self)
```

Comment:
Well, the efficiency will be same but using `self` is the standard way and it will  make your code readable and maintainable. You are basically doing what Python will do implicitly.